### ISSUE TITLE
Mark current scale in Piano Roll not working - LMMS Beta 1.20-RC3

### ISSUE THREAD

**Issue by mundusnine** [LMMS/lmms#3651]
When selecting any scale and right-clicking on piano roll to mark the scale, the scale doesn't get marked in the Piano Roll editor view after clicking mark current scale. Windows version 64 bit.

**Comment by musikBear** [comment]
@mundusnine I think you are doing it wrong.
First: Select a scale in the **scale-dropdown**
Then: Right-click on the **key** that you want for the already chosen scale!
LMMS correctly marks all notes belonging to the the *chosen key* in *your scale*.
I also explain more here:
https://www.youtube.com/watch?v=WhkwmABRtp4

**Comment by zonkmachine** [comment]
I also have an issue with this. It works in 1.1.3 and on stable 1.2.0 it does mark the notes but the marking is all black or very dark so it is a bit too hard to see. @RebeccaLaVie code or theme?

Here is a C major marked.

![major](https://user-images.githubusercontent.com/6368949/27430194-ebdab614-5747-11e7-835d-ac8777767256.png)

**Comment by Umcaruje** [comment]
most probably theme, I'll look into this next week

**Comment by RebeccaLaVie** [comment]
@mundusnine Can you follow the steps posted by MusicBear and send us a screenshot of what the result looks like on your end? I am curious to know if your issue is separate from what zonkmachine has posted.

@zonkmachine You're right, the marked notes are not bright enough. I will create a pull request for this.

**Comment by musikBear** [comment]
@RebeccaLaVie Oh yes -I felt exactly the same as @zonkmachine -first i actually thought iy *was* broken, but then i realixed, that the *marked* notes was those that stood out the *least* :)
How about making then really deep grey (40,40,40)rgb would that look good? Again im always "usage first, design second" -but you know that by now :trollface:

**Comment by mundusnine** [comment]
So it basically gives me the same thing as zonkmachine. The marking works but the color isn't apparent enough, well for me. The devs can decide how they want it in the end but I think it will be easier for users to have a more apparent color like in 1.1.3 version ![scalemarking](https://user-images.githubusercontent.com/29337013/27801338-7562ca56-5feb-11e7-8ebb-99cd1c0b411a.PNG)

**Comment by Hussam-Eddin-Alhomsi** [comment]
I totally agree that marked notes are hard to see. They should be a little bit brighter.

The issue here has to do with the GUI and the UX. The `GUI` and `UX` labels should be added.
The issue has to do with all marked notes `Mark current scale` is working, the issue's title should be changed to: `Marked notes in the Piano Roll are hard to see`.

**Comment by musikBear** [comment]
> I totally agree that marked notes are hard to see. They should be a little bit brighter.

**"They should be a little bit brighter."**
Ehrr.. You do mean **Darker** -right? Current **marked** notes in scale are the **darkest** notes! Not those that are bright. Imo the **marked** notes should be very dark, almost 'black' rgb 80,80,80

**Comment by Sawuare** [comment]
@musikBear 
![image](https://user-images.githubusercontent.com/25503477/29485154-deb51644-84d5-11e7-889d-8bf431235a41.png)
Yes, marked notes should be a little bit brighter. I take back my "hard to see" statement as they are fine for me actually, but a little bit more of brightness would be better.

**Comment by RebeccaLaVie** [comment]
I've already put some thought into this issue.

There are enough colors being used in the song editor grid that finding a combination with enough contrast is very difficult. Changing the marked notes to black as @musikBear mentioned does seem to help (1), but I'm afraid, not enough.

![27430194-ebdab614-5747-11e7-835d-ac8777767256](https://user-images.githubusercontent.com/7960169/31588839-a772756a-b1ac-11e7-9481-08ffc025d0b1.png)

Optimal contrast could be achieved with a transparent overlay (2 and 3), which would require more work to implement, but I do not believe that lighter marked notes will work otherwise.

@Umcaruje

**Comment by musikBear** [comment]
@RebeccaLaVie Would it destroy design if marked notes used blue shades?
Green and blue usually goes well together, but then theres the issue of the color-blind. (jah i just sunk my own idea.. 🤡 )
I then vote for black i think it stands out enough, and im bat-visionary 👀

**Comment by HummusSamurai** [comment]
Is it possible to have a pattern drawn instead of a different color?
That would solve the color-blind issues.

**Comment by musikBear** [comment]
@HummusSamurai 'pattern' like hash-marked -It an idea, but i doubt it will be sufficient clear, and be stress-full for the eyes. I still think black is best as default

**Comment by Sawuare** [comment]
@musikBear I think @HummusSamurai suggests making stamping scales possible, just like how stamping chords is possible.
@HummusSamurai I will open an issue for that, but this issue here should still be fixed.

**Comment by HummusSamurai** [comment]
I believe @musikBear actually understood what I'm going for.

**Comment by raindropsfromsky** [comment]
I will liken this to the children's books for practicing the writing of alphabets.
In those books, they have to move the pencil along a dashed line. 
In other words, the dashed curved line is to guide them.

In LMMS, the marked notes serve the exact same purpose (of guiding the user where to place his notes).

If we look at those books, they have black dashed line on white pages (highest contrast).
In our case, the background is dark gray. 
So why not have a dashed thin **white** line on it as marker?

**Comment by raindropsfromsky** [comment]
Here are some combination of white and gray lines (both **solid** and **fine dashed**).



![image](https://user-images.githubusercontent.com/9047168/35478553-50c47fa4-0406-11e8-9e45-8efc44e6fc5f.png)

![image](https://user-images.githubusercontent.com/9047168/35478558-5f65f614-0406-11e8-829d-d36f0bdcfac5.png)

![image](https://user-images.githubusercontent.com/9047168/35478560-6f035ec2-0406-11e8-88ec-d29e8184bda3.png)
![image](https://user-images.githubusercontent.com/9047168/35478563-783b9dd8-0406-11e8-81c5-2b40f590adc8.png)
![image](https://user-images.githubusercontent.com/9047168/35478564-842d1360-0406-11e8-8f67-425c0ca3132c.png)

![image](https://user-images.githubusercontent.com/9047168/35478569-a330b0c8-0406-11e8-8233-3de4ed0e80b0.png)

Finally, here is a comparison of thick/thin white/gray lines in a single image:
![image](https://user-images.githubusercontent.com/9047168/35478607-83cd7454-0407-11e8-9161-e41ef188ead2.png)

IMHO the **thick white dashed line** stands out against the grid lines.
All the other lines (especially the solid ones) look almost same as the grid lines.

**Comment by musikBear** [comment]
@raindropsfromsky :'p if you had been a lmms 'vet', you would have known, that dashed lines is one of the things that has been disposed of, because it is 
* very disruptive for eyes
* cause headache 
* horrible ugly
Dashed lines gets a HUGE 👎  from me
